client/me/purchases/manage-purchase/index.jsx

```diff
@@ -393,19 +400,41 @@ class ManagePurchase extends Component {
 				'Transfers an existing domain from another provider to WordPress.com, ' +
 					'helping you manage your site and domain in one place.'
 			);
+		} else if ( isJetpackProduct( purchase ) ) {
+			description = (
+				<Fragment>
+					<div className="manage-purchase__description-subcontent">
+						{ getProductDescription( purchase ) }
+					</div>
+					{ isJetpackSearch( purchase ) && (
+						<div className="manage-purchase__description-subcontent">
+							{ translate( '{{strong}}Your Search Tier{{/strong}}: %(tierDescription)s.', {
+								args: {
+									tierDescription: getShortNameCallbackForJetpackSearch( this.props.siteProduct ),
```

Comment:
I'm not sure it's always going to work to get this from the product.

In most cases it will, but there's a potential complication to think about. This page contains renewal info for the subscription, and crucially, that means the price shown on this page is the **renewal** price (the price they would pay if they went through checkout and renewed the subscription now).  This price does not always correspond to their current tier, although it usually will.  The reason it might not is because when the user is close to their auto-renewal/expiration date (and has already received a pre-renewal email quoting a particular renewal price) we are guaranteeing that we won't charge them more than that for that renewal, even if they move to a higher tier in the meantime.

So the scenario to think about would be:
- User had 99 search records ("Up to 100 records" tier) at the time the renewal price was locked, for a cost of $50.
- A few days later, they move up to 101 search records ("Up to 1,000 records" tier).  But their renewal price is still going to be $50, rather than $100.

So it's worth thinking about what to show them in this case. Getting the tier from the product will show them "Up to 1,000 records", which in one sense is accurate, but it doesn't correspond to their renewal price...

I could go ahead and get D41388-code (mentioned in the previous PR, https://github.com/Automattic/wp-calypso/pull/40774) deployed if it will help with this.  In that case, you could get the tier from the purchase object, and in the above scenario it would return "Up to 100 records".  I think that might be what you want, but I'm not sure exactly the best way to display it.  I think we'd like to hide this complication from the user, but also not confuse them about what they're paying for.

Reply:
I had some doubts about my implementation; thanks for the clarification. Your explanation for a potential tier mismatches makes a lot of sense. I agree that adding the current tier info to the purchase object would be a good, straightforward solution our current issue. 

We'll eventually want to figure out how to communicate tier mismatches in the future. Nonetheless, correctly communicating their current tier is a prerequisite before that can happen.

I'll go ahead and review `D41388-code` and work on integrating it with this PR.

Reply:
@DavidRothstein: Would we need to make any changes to this PR now that `D42761-code` has shipped? It sounds like fetching the current tier description from the site product will now work as intended if I understood you correctly from `D41388-code` 😅

Reply:
To get the current tier ("Up to 1,000 records" in the above example) from the site product, this PR actually should have worked all along.  Conceptually, it's probably better to get it from the purchase object (since this page is all about the purchase), and we had talked about modifying D41388-code to make that possible, which could then be used here.

Getting the renewal tier ("Up to 100 records" in the above example) is what **definitely** would require D41388-code, though, since that information is not currently exposed to Calypso anywhere.

Reply:
Gotcha. Once D41388-code is ready, I'll modify this PR to use the purchase object for the current tier details.